flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -887,19 +897,22 @@ public void runIfState(State expectedState, Runnable action, Duration delay) {
 
     // ----------------------------------------------------------------
 
-    private void transitionToState(State newState) {
-        if (state != newState) {
-            LOG.debug(
-                    "Transition from state {} to {}.",
-                    state.getClass().getSimpleName(),
-                    newState.getClass().getSimpleName());
+    @VisibleForTesting
+    <S extends State> void transitionToState(
```

Comment:
why did you change the semantics of "transition to same state" from "ignore" to "fail"?